hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/index.md

```diff
@@ -2029,3 +2029,51 @@ signer for all services.
 For a specific service, the service specific signer is looked up first.
 If that is not specified, the common signer is looked up. If this is
 not specified as well, SDK settings are used.
+
+### <a name="cse"></a> Amazon S3 Client Side Encryption
+Amazon S3 Client Side Encryption(CSE), uses `AmazonS3EncryptionClientV2.java
+` AmazonS3 client. The encryption and decryption is done in AWS SDK side. Atm
```

Comment:
"atm?"

Recommend an intro, features and limitations section:


A key reason this feature (HADOOP-13887) has been unavailable for a long time is that the AWS S3 client pads uploaded objects with a 16 byte footer. This meant that files were shorter when being read that when are listed them through any of the list API calls/getFileStatus(). Which broke many applications, including anything seeking near the end of a file to read a footer, as ORC and Parquet do.

There is now a workaround: compensate for the footer in listings when CSE is enabled.

* when listing files and directories, 16 bytes are subtracted from the length of all non-empty objects.
* directory markers MAY be longer than 0 bytes long.

This "appears" to work; secondly it does in the testing to date. However, the length of files when listed through the S3A client is now going to be shorter than the length of files listed with other clients -including S3A clients where S3-CSE has not been enabled.

features
* Supports client side encryption with keys managed in AWS KMS
* encryption settings propagated into jobs through any issued delegation tokens
* encryption information stored as headers in the uploaded object

Limitations
* performance will be reduced. All encrypt/decrypt is now being done on the client.
* writing files may be slower, as only a single block can be encrypted and uploaded at a time
* Multipart Uploader API disabled